examples/libtorchaudio/speech_recognition/greedy_decoder.py

```diff
@@ -0,0 +1,32 @@
+import torch
+
+
+class Decoder(torch.nn.Module):
+    def __init__(self, labels):
+        super().__init__()
+        self.labels = labels
+
+    def forward(self, logits: torch.Tensor) -> str:
+        """Given a sequence logits over labels, get the best path string
+
+        Args:
+            logits (Tensor): Logit tensors. Shape `[num_seq, num_label]`.
+
+        Returns:
+            str: The resulting transcript
+        """
+        best_path = torch.argmax(logits, dim=-1)  # [num_seq,]
+        prev = ''
+        hypothesis = ''
+        for i in best_path:
+            char = self.labels[i]
+            if char == prev:
```

Comment:
nit: have you tried if `torch.unique_consecutive` improves performance by getting rid of this case?

Reply:
if using `torch.unique_consecutive`, it might turn out faster to use comprehension, e.g.
```python
best_path = torch.argmax(logits, dim=-1)
best_path = torch.unique_consecutive(best_path, dim=-1)
best_path = "".join(self.labels[i] for i in best_path)
best_path = best_path.replace("<s>", "").replace("|", " ")
return best_path
```
or
```python
best_path = torch.argmax(logits, dim=-1)
best_path = torch.unique_consecutive(best_path, dim=-1)
best_path = "".join(self.labels[i] for i in best_path if self.labels[i] != "<s>")
best_path = best_path.replace("|", " ")
return best_path
```

P.S. I find the first option also most readable :)